lib/Rex/Commands/Iptables.pm

```diff
@@ -485,6 +485,49 @@ sub _iptables_list {
   return $ret;
 }
 
+=head2 chain_exists
+
+Returns true if a chain exists in a table, false otherwise.
+
+In this example we create a chain unless it already exists.
+
+ task "create_chain", sub {
+   iptables(t => 'filter', N => 'FOO')
+     unless chain_exists('filter','FOO');
+ };
+
+
+=cut
+
+sub chain_exists {
+  my ( $table, $chain, @params ) = @_;
+  my $iptables = _get_executable( \@params );
+  my @lines    = run "$iptables-save";
+
+  return _chain_exists( $table, $chain, @lines );
```

Comment:
I think it might be simpler and faster to check for the return code of `iptables [--table $table] --list $chain > /dev/null` command. If the chain exists it will be true, but otherwise false.

Perhaps even the built-in `iptables()` sub can be used for that too (but it probably needs to be wrapped in a try-catch block as it `die()`s in case of non-OK return codes).